commons/src/main/java/org/infinispan/commons/marshall/PersistenceContextInitializer.java

```diff
@@ -0,0 +1,25 @@
+package org.infinispan.commons.marshall;
+
+import org.infinispan.commons.dataconversion.MediaType;
+import org.infinispan.commons.io.ByteBufferImpl;
+import org.infinispan.protostream.SerializationContextInitializer;
+import org.infinispan.protostream.annotations.AutoProtoSchemaBuilder;
+
+/**
+ * Interface used to initialise a {@link org.infinispan.protostream.SerializationContext} using the specified Pojos,
+ * Marshaller implementations and provided .proto schemas.
+ *
+ * @author Ryan Emerson
+ * @since 10.0
+ */
+@AutoProtoSchemaBuilder(
+      includeClasses = {
+            ByteBufferImpl.class,
+            MediaType.class,
+            WrappedByteArray.class
```

Comment:
`WrappedByteArray` needs to be written to the store as we need to be able to distinguish between arbitrary user bytes that have been marshalled (UserObject) and bytes which should simply be returned as stored.

Reply:
I've updated MarshalledValueImpl to write `bytes` directly, so `ByteBufferImpl` serialization is no longer necessary.

Reply:
> WrappedByteArray needs to be written to the store as we need to be able to distinguish between arbitrary user bytes that have been marshalled (UserObject) and bytes which should simply be returned as stored.

I was asking because or my pet peeve with the store format changing because of a cache configuration change, as `WrappedByteArray` isn't always the result of the application inserting a `byte[]` in the cache, it's also sometimes because `BinaryEncoder` or `MarshallerEncoder` replaced the application-supplied value with a `byte[]`. I guess that's an encoding problem though, not a marshalling problem.